marlowe-dashboard-client/src/Contract/View.purs

```diff
@@ -684,3 +702,23 @@ getParty (IDeposit _ p _ _) = Just p
 getParty (IChoice (ChoiceId _ p) _) = Just p
 
 getParty _ = Nothing
+
+isParty :: WalletDetails -> MarloweParams -> Party -> Boolean
+isParty walletDetails marloweParams party = Set.member party $ getParties walletDetails marloweParams
```

Comment:
isParty seems to be called a lot and I'm not sure if it changes (or if it changes that much), we might want to have `getParties` cached in the contract state, right?

Reply:
Good point. In the real thing, we might let people give their roles to others. But for now at least, it's never going to change.